pkg/build/types/parser/deffile.go

```diff
@@ -131,51 +132,103 @@ func getSectionName(line string) string {
+// parseTokenSection into appropriate components to be placed into a types.Script struct
+func parseTokenSection(tok string, sections map[string]*types.Script, files *[]types.Files) error {
 	split := strings.SplitN(tok, "\n", 2)
 	if len(split) != 2 {
 		return fmt.Errorf("Section %v: Could not be split into section name and body", split[0])
 	}
 
 	key := getSectionName(split[0])
+
+	// parse files differently to allow multiple files sections
+	if key == "files" {
+		f := types.Files{}
+		sectionSplit := strings.SplitN(strings.TrimLeft(split[0], "%"), " ", 2)
+		if len(sectionSplit) == 2 {
+			f.Args = sectionSplit[1]
+		}
+
+		// Files are parsed as a map[string]string
+		filesSections := strings.TrimSpace(split[1])
+		subs := strings.Split(filesSections, "\n")
+		for _, line := range subs {
+			if line = strings.TrimSpace(line); line == "" || strings.Index(line, "#") == 0 {
+				continue
+			}
+			var src, dst string
+			lineSubs := strings.SplitN(line, " ", 2)
+			if len(lineSubs) < 2 {
+				src = strings.TrimSpace(lineSubs[0])
+				dst = ""
+			} else {
+				src = strings.TrimSpace(lineSubs[0])
+				dst = strings.TrimSpace(lineSubs[1])
+			}
+			f.Files = append(f.Files, types.FileTransport{Src: src, Dst: dst})
+		}
+
+		// look through existing files and append to them if they already exist
+		for i, ef := range *files {
+			if ef.Args == f.Args {
+				ef.Files = append(ef.Files, f.Files...)
+				// replace old file struct with newly appended one
+				(*files)[i] = ef
+				return nil
+			}
+		}
+
+		*files = append(*files, f)
+		return nil
+	}
+
 	if appSections[key] {
 		sectionSplit := strings.SplitN(strings.TrimLeft(split[0], "%"), " ", 3)
 		if len(sectionSplit) < 2 {
 			return fmt.Errorf("App Section %v: Could not be split into section name and app name", sectionSplit[0])
 		}
 
 		key = strings.Join(sectionSplit[0:2], " ")
+		// create app script pbject to populate
+		if _, ok := sections[key]; !ok {
+			sections[key] = &types.Script{}
+		}
+	} else {
+		// create section script object if its a non-standard section
+		if _, ok := sections[key]; !ok {
+			sections[key] = &types.Script{}
+		}
+		sectionSplit := strings.SplitN(strings.TrimLeft(split[0], "%"), " ", 2)
+		if len(sectionSplit) == 2 {
+			sections[key].Args = sectionSplit[1]
+		}
 	}
 
-	sections[key] += split[1]
-
+	sections[key].Script += split[1]
 	return nil
 }
 
 func doSections(s *bufio.Scanner, d *types.Definition) error {
-	sectionsMap := make(map[string]string)
-
+	sectionsMap := make(map[string]*types.Script)
+	files := []types.Files{}
 	tok := strings.TrimSpace(s.Text())
 
 	// skip initial token parsing if it is empty after trimming whitespace
 	if tok != "" {
-		//check if first thing parsed is a header/comment or just a section
+		// check if first thing parsed is a header/comment or just a section
 		if tok[0] != '%' {
 			if err := doHeader(tok, d); err != nil {
 				return fmt.Errorf("failed to parse DefFile header: %v", err)
```

Comment:
I dont think error messages are supposed to have any caps...

```suggestion
 				return fmt.Errorf("failed to parse deffile header: %v", err)
```